backpack/core/derivatives/adaptive_avg_pool_nd.py

```diff
@@ -27,7 +28,11 @@ def check_parameters(
         Raises:
             NotImplementedError: if the given shapes do not match
         """
-        if ("cuda" in str(module.input0.device)) and (self.N == 3):
+        if (
+            TORCH_VERSION_HIGHER_THAN_1_9_1 is False
+            and ("cuda" in str(module.input0.device))
```

Comment:
Use `module.input0.is_cuda` (reference:https://discuss.pytorch.org/t/how-can-i-know-if-a-variable-is-a-cuda-varialbe/8460)